### PR TITLE
MF-222 MF-221 MF-228 MF-219 MF-220 MF-214

### PR DESCRIPTION
**Pull request by dashton82**
Journey for employer from add apprentice after creating a cohort

### CODE REVIEW

File: src/SFA.DAS.Reservations.Application/Reservations/Commands/CacheReservationEmployer/CacheReservationEmployerCommandValidator.cs [hunk 1]
```diff
@@ -35,12 +35,19 @@ public async Task<ValidationResult> ValidateAsync(CacheReservationEmployerComman
             }
             else
             {
-                var globalRules = await _rulesService.GetAccountFundingRules(command.AccountId);
-                if (globalRules.GlobalRules.Any(c => c != null && c.RuleType == GlobalRuleType.ReservationLimit) &&
-                    globalRules.GlobalRules.Count(c => c.RuleType == GlobalRuleType.ReservationLimit) > 0)
+                var accountFundingRulesApiResponse = await _rulesService.GetAccountFundingRules(command.AccountId);
+                if (accountFundingRulesApiResponse.GlobalRules.Any(c => c != null && c.RuleType == GlobalRuleType.ReservationLimit) &&
+                    accountFundingRulesApiResponse.GlobalRules.Count(c => c.RuleType == GlobalRuleType.ReservationLimit) > 0)
```
Comment (robinoconnor) [inline src/SFA.DAS.Reservations.Application/Reservations/Commands/CacheReservationEmployer/CacheReservationEmployerCommandValidator.cs]: Isn't the bottom check asking the same thing as the top one?

Top asks if there are any global rules that are not null and have a type of reservation limit
bottom one asks for the number of rules with the same type and then said is there more than zero.

File: src/SFA.DAS.Reservations.Application/Reservations/Commands/CacheReservationEmployer/CacheReservationEmployerCommandValidator.cs [hunk 2]
```diff
@@ -35,12 +35,19 @@ public async Task<ValidationResult> ValidateAsync(CacheReservationEmployerComman
...
                 {
                     result.FailedRuleValidation = true;
                 }
+
+                var globalRulesApiResponse = await _rulesService.GetFundingRules();
+                if (globalRulesApiResponse.GlobalRules != null && globalRulesApiResponse.GlobalRules.Any(c => c != null && c.RuleType == GlobalRuleType.FundingPaused) &&
+                    globalRulesApiResponse.GlobalRules.Count(c => c.RuleType == GlobalRuleType.FundingPaused) > 0)
```
Comment (robinoconnor) [inline src/SFA.DAS.Reservations.Application/Reservations/Commands/CacheReservationEmployer/CacheReservationEmployerCommandValidator.cs]: same as the one above. is the bottom check doing the same as the one before it?

File: src/SFA.DAS.Reservations.Web.UnitTests/Models/WhenConstructingAReservationViewModel.cs [hunk 3]
```diff
@@ -59,11 +63,12 @@ public class WhenConstructingAReservationViewModel
         public void And_Course_Is_Null_Then_Sets_CourseDescription_To_Unknown(
             Reservation reservation,
             string url,
-            string alephid)
+            string accountHashedId,
+            string accountLegalEntityPublicHashedId)
```
Comment (robinoconnor) [inline src/SFA.DAS.Reservations.Web.UnitTests/Models/WhenConstructingAReservationViewModel.cs]: In this file a number of accountHashedIds and accountLegalEntityPublicHashedId do not seem to be used but are in the parameters. Can they be removed?

File: src/SFA.DAS.Reservations.Web/Controllers/EmployerReservationsController.cs [hunk 4]
```diff
@@ -228,13 +247,15 @@ public async Task<IActionResult> PostSelectCourse(ReservationsRouteModel routeMo
                 await _mediator.Send(new CacheReservationCourseCommand
                 {
                     Id = routeModel.Id.Value,
-                    CourseId = selectedCourseId
+                    CourseId = selectedCourseId,
```
Comment (robinoconnor) [inline src/SFA.DAS.Reservations.Web/Controllers/EmployerReservationsController.cs]: does this need a trailing comma given its the last variable being set?

File: src/SFA.DAS.Reservations.Application/Reservations/Commands/CacheReservationEmployer/CacheReservationEmployerCommandValidator.cs (hunk 1 above)
Comment (dashton82) [reply src/SFA.DAS.Reservations.Application/Reservations/Commands/CacheReservationEmployer/CacheReservationEmployerCommandValidator.cs]: they are different, one is checking global rules, the other account rules

**Pull request merged**